### ISSUE TITLE
Some crashes on differents contexts

### ISSUE THREAD

**Issue by Messius58** [phase1geo/Minder#260]
Hi,

I use to minder on linux for create japanese grammar mind mapping, It's the most simple and efficient that i have to until now.
Only i have some problems with it.

.1 The japanese characters show into different place on the map (up / right, left side, ...) but not on the line of text.
.2 The soft crash often when i delete or modify all characters on the second line of text.
.3 It would be nice to do a temporary backup of the current file or not to automatically backup, but on ctrl+s, to avoid loss of all written information ;-(

I don't know vala, i can't to help you on the debugging, sorry.

Thanks

**Comment by phase1geo** [comment]
What version of Minder do you have installed? What distribution of Linux are you using?

**Comment by Messius58** [comment]
I use the latest version 1.11.3 on opensuse tumbleweed with kde plasma 5.20.
My UI is french and i use a japanese keyboard 109 keys with ibus and ibus-anthy.

**Comment by phase1geo** [comment]
Do you know what version of Gtk your system is using?

**Comment by Messius58** [comment]
The libgtk3 is 3.24 and libcairo is 1.16. 
I will launch it with gdb to have more information.

**Comment by phase1geo** [comment]
Could you also add a screenshot of a tree that lays out the text incorrectly?

**Comment by Messius58** [comment]
I join a screenshot with japanese character on the first blue line when i wrote in the new primary node at the bottom.
The position of text is showing randomly.
![minder-japanese](https://user-images.githubusercontent.com/59862077/102727545-85c77300-4369-11eb-8d16-9c966cc3826b.png)

**Comment by Messius58** [comment]
I attach a file with the output of gdb when the application crash when i modify a text on the second line with undo and either alphabetic or either japaneses character. In fact the two first character not show and if undo immediatly, the soft freeze and when i push on escape or enter it crash.


[minder-gdb.txt](https://github.com/phase1geo/Minder/files/5721721/minder-gdb.txt)

**Comment by phase1geo** [comment]
I think this text window is just the input method box. I haven't found a way to control where the library displays this box unfortunately. This special input method box is only used for a handful of languages. Most languages don't require this input box so that is why the input text is displayed in the map where it belongs. I believe that once you have committed the text in the input box, it should be displayed at the correct position in the map (which is what your diagram is showing me).

I don't think that I can fix this specific issue currently.

**Comment by phase1geo** [comment]
The GDB file is missing some information. After the segmentation fault occurs and you see the gdb prompt, type the command 'where' and include that information.  That will contain the entire stack trace which will help me diagnose the issue.  Could you generate and append that information to this bug report?

**Comment by phase1geo** [comment]
As for the issue with the sidebar taking up the whole screen -- on the command-line, could you enter the following command in a terminal and let me know what value you see?

gsettings get com.github.phase1geo.minder properties-width

**Comment by Messius58** [comment]
> The GDB file is missing some information. After the segmentation fault occurs and you see the gdb prompt, type the command 'where' and include that information. That will contain the entire stack trace which will help me diagnose the issue. Could you generate and append that information to this bug report?

Below the debug file.
[minder-gdb.txt](https://github.com/phase1geo/Minder/files/5721893/minder-gdb.txt)

**Comment by Messius58** [comment]
> As for the issue with the sidebar taking up the whole screen -- on the command-line, could you enter the following command in a terminal and let me know what value you see?
> 
> gsettings get com.github.phase1geo.minder properties-width

I have this :

**Comment by phase1geo** [comment]
Trying running the following command:

gsettings set com.github.phase1geo.minder properties-width 300

**Comment by phase1geo** [comment]
As for the GDB file, this latest file is basically the same as the first.  What is the command that you are using to run Minder under gdb?

When an application gets a segfault when running in gdb, the gdb prompt will be displayed in the terminal allowing you get debug the segfault.  When this prompt is displayed in the terminal, you can run the command 'where' to see the current stack where the failure is occurring. The output from the 'where' command is what I want to see (I'm not really interested in the standard output that is being sent to the terminal as you are showing me).

**Comment by Messius58** [comment]
Effectively, a problem of copy between the terminal and kate.
I'm sorry.

[minder-gdb-where-2.txt](https://github.com/phase1geo/Minder/files/5721987/minder-gdb-where-2.txt)

I also tried with others apps gnome (Grisbi, rhythmbox) is work as expected. Perhaps a problem in the worklow to get the input keyword : Input direct method ? HAL method ? ...

**Comment by phase1geo** [comment]
I'm assuming that last comment was in regards to the placement of the input method window? If so, it is not surprising that most input cases show the window in the proper location.  Gtk entry and text widgets support input method windows natively, displaying them somewhere within the text widget (Minder's note field uses a Gtk textview widget where you should notice the correct/expected placement of the input method window).

The difference is that Minder does not use a Gtk entry or text widget to display/enter text. It is displaying text within a DrawingArea widget (a canvas widget) using Pango to display the text.  Unfortunately, Gtk DrawingArea widgets do not support input method windows natively and Gtk doesn't provide API support for placing input method windows at a given X,Y position within a widget (at least not as of Gtk 3.22 which is what I need to use in elementary OS for the time being until Odin is released).  As such, I can't control the position of this box. I will take another look at this when I can update to Gtk 3.24 and see if I can get this issue fixed, but until then, there's nothing I can do about it.

**Comment by phase1geo** [comment]
As for the GDB file, unfortunately you are attempting to run with a version of Minder that was not compiled with debug symbols compiled in so the stack trace isn't useful. Is there any way you would be willing to download Minder 1.11.3 from source (from GitHub), build it and run with the './app debug' command in the main Minder directory? This will compile and run with debugging enabled within GDB.

**Comment by Messius58** [comment]
gsettings set com.github.phase1geo.minder properties-width 300 not working, the application restore the value 1905.

**Comment by phase1geo** [comment]
So if Minder is not running, the command is executed, and Minder is restarted, Minder is changing this value back to 1905?

**Comment by Messius58** [comment]
Yes, at least two times. Only in the code of minder, i don't see of property setter, it's strange. Actually the value is keeping, perhaps after a crash the properties are lost in the conf file and get the value 0 for sidebar_width in the source file "DrawArea" line 668.

**Comment by phase1geo** [comment]
I have committed some changed to the devel branch that will (hopefully) address the issue with the sidebar. I would appreciate it if you could pull the latest devel branch and see if the problem still exists.

If you are able to run this version, I would still be interested in getting the stacktrace information for the segfault issue (assuming it still exists in the devel branch).

**Comment by Messius58** [comment]
I need a valid favorites.xml for execute the debug version of program with gdb.
Into .local/share/minder, i have a image directory and a tab_states.xml file.

**Comment by phase1geo** [comment]
When running in gdb, just enter the 'c' command to continue running. You may have to do that a few times until you get to the segfault.

**Comment by Messius58** [comment]
I have already done and continue twice, the application is not showing and no errors in gdb.
困ってしまった ;-(
I attach the gdb output.
[minder-gdb-debug.txt](https://github.com/phase1geo/Minder/files/5727416/minder-gdb-debug.txt)

For instant i use the debug version without gdb, and no crash, the properties tabs are 300.

**Comment by Messius58** [comment]
I patched the source file to bypass the favorites.xml.
I think that the problem is with deleting text in the node which uses the undo function, but sometimes first two characters no longer appear on the second line when I write / delete multiple times.
[canvas-bug.txt](https://github.com/phase1geo/Minder/files/5728543/canvas-bug.txt)

**Comment by phase1geo** [comment]
Do you have .minder file that I could use to replicate this problem?

**Comment by Messius58** [comment]
I just wanted to save a file and reload it, but unfortunately the file remains correct. The problem comes when you write 2 lines. I place the cursor at the end of the second line and I go up with the arrows of the keyboard to the beginning of the first line. Once this selection is made, I press del to erase and again type a text. that's when it crashes because the pointer to name.text.text on the function delete is segmentation fault.

I wrote a new way to modify the text of a node to avoid the problems that could come from unicode. It works but I would like to make it even better. As soon as I'm done, i push a new request.

**Comment by Messius58** [comment]
This solution also solves the problem that characters appear anywhere on the screen when I write in Japanese.

**Comment by phase1geo** [comment]
Okay, I will wait for the pull request then.
